onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -130,16 +130,20 @@ bool ReshapeFusion::Fuse_Subgraph1(Node& reshape, Graph& graph, const logging::L
     }
     // Try to find path [Root] --> Shape --> Gather(indices=i) --> Unsqueeze (axes=0) --> Concat [input i]
     bool matched = ReshapeFusion::Fuse_Subgraph2(graph, root_input, concat, i, shape_value, logger);
-    const Node* p_cur_node = graph_utils::GetInputNode(concat, i);
+    const NodeArg* concat_input_node_arg = concat.InputDefs()[i];
     if (matched) {
       shape_value.push_back(0);
-    } else if (p_cur_node != nullptr) {
```

Comment:
Not sure if this check to establish that the Concat node has an upstream node is needed (?). Can add this back if needed.